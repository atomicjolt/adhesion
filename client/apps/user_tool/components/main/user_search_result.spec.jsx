import React from 'react';
import { shallow } from 'enzyme';

import UserSearchResult from './user_search_result';

describe('UserSearchResult', () => {
  const user = {
    sortable_name: 'Washington, George',
    email: 'countryfather@revolution.com',
    roles: ['admin', 'teacher'],
    login_id: 'countryfather@revolution.com',
    sis_user_id: 'george_123',
  };

  it('renders the user as a search result', () => {
    const result = shallow(<UserSearchResult user={user} />);

    expect(result).toMatchSnapshot();
  });

  describe('when the edit user button is clicked', () => {
    it('opens the edit user modal', () => {
      const userSearchResult = shallow(<UserSearchResult user={user} />);

      let modalComponent = userSearchResult.find('EditUserModal');
      expect(modalComponent.prop('isOpen')).toBe(false);

      userSearchResult.find('button').simulate('click');

      modalComponent = userSearchResult.find('EditUserModal');
      expect(modalComponent.prop('isOpen')).toBe(true);
    });
  });
});
