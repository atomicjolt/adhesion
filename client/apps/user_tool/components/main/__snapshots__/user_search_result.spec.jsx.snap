// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSearchResult renders the user as a search result 1`] = `
<Fragment>
  <tr>
    <td>
      <button
        onClick={[Function]}
        type="button"
      >
        Washington, George
      </button>
    </td>
    <td>
      countryfather@revolution.com
    </td>
    <td>
      admin
      teacher
    </td>
    <td>
      countryfather@revolution.com
    </td>
    <td>
      george_123
    </td>
  </tr>
  <EditUserModal
    closeModal={[Function]}
    isOpen={false}
    userToEdit={
      Object {
        "email": "countryfather@revolution.com",
        "login_id": "countryfather@revolution.com",
        "roles": Array [
          "admin",
          "teacher",
        ],
        "sis_user_id": "george_123",
        "sortable_name": "Washington, George",
      }
    }
  />
</Fragment>
`;
